src/test/java/edu/hm/hafner/analysis/parser/EclipseMavenParserTest.java

```diff
@@ -96,20 +100,32 @@ void issue55368() {
                     .hasSeverity(Severity.WARNING_NORMAL)
                     .hasLineStart(9)
                     .hasMessage("The method isValid(String) from the type X can potentially be declared as static")
-                    .hasFileName("/home/piotr/.../X.java");
+                    .hasFileName("/home/piotr/.../X.java")
+                    .hasCategory("Code");
             softly.assertThat(warnings.get(1))
                     .hasSeverity(Severity.WARNING_NORMAL)
                     .hasLineStart(54)
                     .hasMessage("The method as10(String) from the type X can potentially be declared as static")
-                    .hasFileName("/home/piotr/.../X.java");
+                    .hasFileName("/home/piotr/.../X.java")
+                    .hasCategory("Code");
             softly.assertThat(warnings.get(2))
                     .hasSeverity(Severity.WARNING_NORMAL)
                     .hasLineStart(60)
                     .hasMessage("The method as13(String) from the type X can potentially be declared as static")
-                    .hasFileName("/home/piotr/.../X.java");
+                    .hasFileName("/home/piotr/.../X.java")
+                    .hasCategory("Code");
```

Comment:
Extract "Code" to constant in EclipseParser so we can change that later more easily. (Maybe someone would like to add some other categories as well)

Reply:
I can put the test string into a constant for the JUnits.

But, what is the overall goal for Categories, should each parser be generating their own based on the categories the tool it's self generates, or should the tools try to course-grain group them?  Or is grouping something that would be done by warnings-ng directly?

Reply:
I am not sure what makes more sense here. If you are using a single tool then the category provided by the single tool probably is the best option. However, in Java we have a lot of similar tools like PMD and CheckStyle and so on. They currently provide different categories, but is this really helpful for the user? I think a good way would be for my plugin to provide a set of predefined categories (e.g. the categories of PMD). Then each parser can map its warnings to one of these categories (as this is done with severities). Especially for parsers of tools that have no category at all (Eclipse) it would make sense to use a set of categories of my plugin. This is something SonarQube or Codacy is doing (https://app.codacy.com/project/uhafner/warnings-ng-plugin/dashboard). Even if we do not map tools that provide an individual category it would make sense to provide at least a set of possible categories. This would help in the UI to provide a pie chart with categories (e.g. there are only six different ones available). Filtering and grouping is done by the warnings plugin.

Reply:
So should I hold off on this till the defined set of categories is configured?

Or would another layer be useful, categories and category groups. So individual parser can still generate their categories based on the source tool, but they can also group them up, for the UI pie chart support?

So for example:

Defined group == Documentation
Eclipse == Javadoc catagory --> Documentation group
Checkstyle == One of it's inline comment categories --> Documentation group

Reply:
I think adding a defined set of categories is something I will add in the future as an additional layer. Maybe this is something that could be done in the warnings-ng plugin only during the rendering. Then no information of the original parser is lost.

So use the suggested category "JavaDoc" in your parser.

Reply:
Will do.